src/compas_rhino/geometry/brep/edge.py

```diff
@@ -131,28 +142,44 @@ def is_ellipse(self):
 
     def _get_curve_geometry(self):
         curve = self._edge.EdgeCurve
-        if self.is_line:
-            type_ = "line"
-            curve = curve_to_compas_line(curve)
-        # TODO: there is an edge/trim direction issue when creating and edge from circle
-        # elif self.is_circle:
-        #     type_ = "circle"
-        #     curve = curve_to_compas_circle(curve)
-        # elif self.is_ellipse:
-        #     type_ = "ellipse"
-        #     curve = curve_to_compas_ellipse(curve)
-        else:
-            type_ = "nurbs"
-            curve = self._curve
-        return type_, curve
+        domain = [self._edge.Domain[0], self._edge.Domain[1]]
+        _, frame = curve.FrameAt(0)
+        if isinstance(curve, LineCurve):
+            return "line", curve_to_compas_line(curve), frame, domain
+        if isinstance(curve, NurbsCurve):
+            return "nurbs", RhinoNurbsCurve.from_rhino(curve), frame, domain
+        if isinstance(curve, ArcCurve):
+            if not curve.IsClosed:
+                return "arc", arc_to_compas(curve.Arc), curve.Arc.Plane, domain
+            is_circle, circle = curve.TryGetCircle()
+            if is_circle:
+                return "circle", circle_to_compas(circle), circle.Plane, domain
+            is_ellipse, ellipse = curve.TryGetEllipse()
+            if is_ellipse:
+                return "ellipse", ellipse_to_compas(ellipse), ellipse.Plane, domain
+            return "nurbs", curve.ToNurbsCurve(), frame, domain
+        raise ValueError("Unknown curve type: {}".format(curve.__class__.__name__))
 
     @staticmethod
-    def _create_curve_from_data(curve_type, curve_data):
+    def _create_curve_from_data(curve_type, curve_data, frame_data, domain):
+        frame = Frame.from_data(frame_data)
         if curve_type == "line":
-            return line_to_rhino_curve(Line.from_data(curve_data))
+            line = Line.from_data(curve_data)
+            curve = LineCurve(line_to_rhino(line))
         elif curve_type == "circle":
-            return circle_to_rhino_curve(Circle.from_data(curve_data))
+            circle = circle_to_rhino(Circle.from_data(curve_data))
+            circle.Plane = frame_to_rhino_plane(frame)
+            curve = ArcCurve(circle)
         elif curve_type == "ellipse":
-            return ellipse_to_rhino_curve(Ellipse.from_data(curve_data))
+            ellipse = ellipse_to_rhino(Ellipse.from_data(curve_data))
+            ellipse.Plane = frame_to_rhino_plane(frame)
+            curve = NurbsCurve.CreateFromEllipse(ellipse)
+        elif curve_type == "arc":
+            arc = arc_to_rhino(Arc.from_data(curve_data))
+            curve = ArcCurve(arc)
+        elif curve_type == "nurbs":
+            curve = RhinoNurbsCurve.from_data(curve_data).rhino_curve
         else:
-            return RhinoNurbsCurve.from_data(curve_data).rhino_curve
+            raise ValueError("Unknown curve type: {}".format(curve_type))
```

Comment:
there is no parabola?

Reply:
Represented in Rhino as `NurbsSurface` as far as could find.
https://developer.rhino3d.com/api/RhinoCommon/html/M_Rhino_Geometry_NurbsCurve_CreateParabolaFromVertex.htm